### PR TITLE
PS-269: Fix main.log_options_cmdline --sanitize

### PR DESCRIPTION
**Pull request by inikep**
Change `log_error_verbosity` to `sys_var::PARSE_EARLY`
as my_init_signals() was moved before `sys_var::PARSE_NORMAL`
and my_init_signals() allows to use setrlimit() which can generate LogErr()
that depends on `log_error_verbosity`.

### CODE REVIEW

**Review by laurynas-biveinis — APPROVED** [review]
Please file upstream

**Comment by inikep** [comment]
Pushed a completely rewritten solution that changes `log_error_verbosity` to `sys_var::PARSE_EARLY`.

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by inikep** [comment]
Fixed issue with printing warnings with `mysqld --help`.

File: sql/mysqld.cc [hunk 1]
```diff
@@ -5763,12 +5763,14 @@ int mysqld_main(int argc, char **argv)
   init_pfs_instrument_array();
 #endif /* WITH_PERFSCHEMA_STORAGE_ENGINE */
 
+  /* init_error_log() is required by error_log_printf() in
+   * option_error_reporter() */
```
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: nit: why 2nd line started with *

File: sql/mysqld.cc [hunk 2]
```diff
@@ -6969,6 +6971,52 @@ static bool read_init_file(char *file_name) {
...
+
+  /* create a copy of remaining_argv */
+  int copy_argc = remaining_argc;
+  char **copy_argv = new char *[copy_argc + 1];
+  for (int i = 0; i < copy_argc; i++) copy_argv[i] = remaining_argv[i];
+  copy_argv[copy_argc] = nullptr;
+
+  /* select only "help", "initialize", "initialize-insecure" options */
+  vector<my_option> init_options;
+  for (my_option *opt = my_long_early_options; opt->name != NULL; opt++) {
```
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: nullptr

File: sql/mysqld.cc [hunk 3]
```diff
@@ -6969,6 +6971,52 @@ static bool read_init_file(char *file_name) {
...
+    if (!strcmp("help", opt->name) || !strcmp("initialize", opt->name) ||
```
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: Maybe add some static std::vector<string> foo = {"help", ... }; std::find(foo.cbegin(), foo.cend(), opt->name);

File: sql/mysqld.cc [hunk 4]
```diff
@@ -6969,6 +6971,52 @@ static bool read_init_file(char *file_name) {
...
+  Process command line options but use only "help", "initialize",
+  "initialize-insecure". If one of these options exists then change default
+  value of log_error_verbosity.
+*/
+static void adjust_log_error_verbosity(vector<my_option> *all_early_options) {
+  if (remaining_argc <= 1) return;
+
+  /* create a copy of remaining_argv */
+  int copy_argc = remaining_argc;
+  char **copy_argv = new char *[copy_argc + 1];
```
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: std::vector<char **>? handle_options should be able to take &copy_argv[0]

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by inikep** [comment]
The commit is updated to address the comments.

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**